sos/plugins/vdsm.py

```diff
@@ -0,0 +1,153 @@
+# Copyright (C) 2018 Red Hat, Inc.
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+
+import ast
+import os
+import stat
+import re
+
+from sos.plugins import Plugin, RedHatPlugin
+
+
+# This configuration is based on vdsm.storage.lvm.LVM_CONF_TEMPLATE.
+#
+# locking_type is set to 0 in order to match lvm sos commands. With this
+# configuration we don't take any locks, so we will never block because
+# there is a stuck lvm command.
+# locking_type=0
+#
+# use_lvmetad is set to 0 in order not to show cached, old lvm metadata.
+# use_lvmetad=0
+#
+# preferred_names and filter config values are set to capture Vdsm devices.
+# preferred_names=[ '^/dev/mapper/' ]
+# filter=[ 'a|^/dev/mapper/.*|', 'r|.*|' ]
+LVM_CONFIG = """
+global {
+    locking_type=0
+    use_lvmetad=0
+}
+devices {
+    preferred_names=[ "^/dev/mapper/" ]
+    ignore_suspended_devices=1
+    write_cache_state=0
+    disable_after_error_count=3
+    filter=[ "a|^/dev/mapper/.*|", "r|.*|" ]
+}
+"""
+LVM_CONFIG = re.sub(r"\s+", " ", LVM_CONFIG).strip()
+
+
+class Vdsm(Plugin, RedHatPlugin):
+    """vdsm Plugin"""
+
+    packages = (
+        'vdsm',
+        'vdsm-client',
+    )
+
+    plugin_name = 'vdsm'
+
+    def setup(self):
+        os.environ["LVM_SYSTEM_DIR"] = "/var/run/vdsm/lvm"
```

Comment:
It may be necessary to remove this from `os.environ` at the end of `setup()`: today plugins can run in parallel, and in any order, which means that if the LVM2 plugin ran after `vdsm` it would pick up this environment setting.

Assuming that's not wanted then I think it's the best option (even if VDSM deployments don't use the `lvm2` plugin, we should try not to interfere). If that is the case then we may need to think about a better way to coordinate them.